### DISCUSSION TITLE
[BUG] Plotting stuck at beginning of phase 4

### DISCUSSION THREAD

**Discussion by huiui** [Chia-Network/chia-blockchain#3228] [General]
Plotting often stuck at beginning of phase 4, sometime for hours that I have to restart GUI. When is stuck, all other parallel plotting process will be stuck as well, regardless which phase they are in.

Starting phase 4/4: Write Checkpoint tables into "/Volumes/Passport_A/Temp/plot-k32-2021-04-29-14-48-0cc8a805a6873981d9c3dc5e2703848b1dbba95f8d965373af25914777c98986.plot.2.tmp" ... Fri Apr 30 05:00:48 2021
	Starting to write C1 and C3 tables
	Bucket 0 uniform sort. Ram: 3.251GiB, u_sort min: 1.125GiB, qs min: 0.281GiB.

**Comment by sargonas** [comment]
based on your volume path, it appears you have a WD Passport drive? I've seen several threads about those drives having aggressive sleep modes that can mess things up some, so that might be your issue?

If you have the capability, can you try writing plots out to a different final destination drive to see if the issue persists?

(Also moving this over to the Discussions area, if this ends up resulting in a clear case of a software issue/bog we can always bring it back here as a Bug)

**Comment by huiui** [comment]
Yes, I am using a WD Passport drive.
Should I try a different temp drive or a different final destination drive?

**Comment by sargonas** [comment]
I would try that, at least at first to see if it works or not. If it works fine on a different drive, then the issue is likely the sleep mode on your passport drive. If you still have the issue, that at least rules out the sleep mode as the culprit.

**Reply by huiui** [reply]
Tried with a different system and with a different drive (usb-c connected external PCIE SSD -- WD SN550 NVMe), and using command line plotting this time.  Same issue is happening... Stuck at beginning of phase 4

I am parallelly doing 3 plotting, when one stuck, the other two also stopped plotting, even though they have not reached phase 4 yet.

Update: it stuck for 30 mins, and now it resumed plotting now...  is this a normal behavior?

**Comment by huiui** [comment]
This issue still exist. Happened on multiple system/SSD drive. Some time it only stuck at beginning of phase 4 for 10 mins, sometime for hours. When it is happening, it significantly slow down other parallel plotting process and CPU load decreased to almost zero.

**Comment by ReneSmeekes** [comment]
I'm seeing the same problem on a system that was plotting just fine until yesterday. This is now the second time the process is stuck at.

```
Starting phase 4/4: Write Checkpoint tables into "C:\\ChiaTemp\\plot-k32-2021-05-01-23-33-68b070d640a96aa94e4bf73fbeae6a3c7e4b7d80c67ac82df4b65e1208b0ea6f.plot.2.tmp" ... Sun May  2 11:21:27 2021
	Starting to write C1 and C3 tables
	Bucket 0 uniform sort. Ram: 4.381GiB, u_sort min: 1.125GiB, qs min: 0.281GiB.
```
Since my temp location is on the system drive I highly doubt the sleeping can be the issue.

Edit: While this happens the chia.exe process responsible for the plotting seems to use no IO whatsoever. The IO counts and KB's don't change at all. But systemwide performance grinds to a halt. Task manager says there is 100% disk usage, but lists no process that is actually using that disk. Fairly confusing. The SSD is a Samsung SSD 850 EVO. Yesterday it was stuck for so long and seemed to be doing nothing that I broke off the process and started over (Which kind of sucks at 94%). Today it seems it does eventually move on. What's interesting is that I'm plotting on another system where it just blazes through this step. Doesn't take more than a few seconds to move on.

**Reply by hydrazine1209** [reply]
Same issue here. Do you have a solution for this thanks

**Comment by imedjendoubi** [comment]
Reporting the same problem here, plotting is stuck on phase 4 (94 %) for hours now

Temp and final destination are on two separate hard drives

**Comment by huiui** [comment]
So this is likely a software issue? Should we move it back to Issue area?

**Comment by Pterko** [comment]
I have the same issue. The plotting process stuck at Phase 4 with 100% CPU load on one core. I'm not using a WD Passport drive, I'm using a regular SSD.
<details>
  <summary>Screenshots</summary>

![image](https://user-images.githubusercontent.com/3855974/116872205-326e0700-ac1e-11eb-90a4-64d224046789.png)
![image](https://user-images.githubusercontent.com/3855974/116872234-3ac64200-ac1e-11eb-9ca3-13994885f1b4.png)
</details>

**Comment by iXavierLiu** [comment]
![image](https://user-images.githubusercontent.com/31836909/116882832-78938c80-ac57-11eb-9e13-4486f1f66acf.png)
![image](https://user-images.githubusercontent.com/31836909/116882966-a2e54a00-ac57-11eb-8304-df80844aedaf.png)

i have the same issue, CPU and memory usage are not lost, but the hard disk I/O speed is zero, it seems to be stuck.

**Reply by iXavierLiu** [reply]
I solved the pssd stuck problem by turning off the usb device hibernation in the windows power options
![image](https://user-images.githubusercontent.com/31836909/117203189-50f02000-ae21-11eb-980d-d050f809536b.png)

**Reply by iXavierLiu** [reply]
@huiui

**Reply by huiui** [reply]
Wow! Thank you so much, I will give it a try.

**Comment by huiui** [comment]
Did anyone found work around for this issue?  I parallel plot 8 at the same time, when one get stuck at 94% , I have to restart the computer and lost all plots. Happened a few times last week, really frustrating.

**Reply by imedjendoubi** [reply]
Version 1.1.1 solved everything for me, give it a try

**Reply by huiui** [reply]
Thank you, I will try it.

**Comment by chianew** [comment]
I got this problem serveral times, it cost hours. Ithink it should be a bug

**Comment by nonogrub** [comment]
Back to 1.1.2 and all my problems gone ;)

**Reply by imedjendoubi** [reply]
I'll give it a try, thanks

**Comment by IEthMiner** [comment]
I also have this issue on Windows 10, in this instance it's been here for 20ish mins though have had it sticking around here for over 4 hours ! 

![image](https://user-images.githubusercontent.com/42719736/118507186-e9da4180-b725-11eb-883b-c0ef6b33d717.png)

Temp Directory is 2TB NVME plotting to External Drive - I've changed every setting possible to ensure the HD does not sleep (I use these exact hard drives on other computers and have no issues)

**Comment by hydrazine1209** [comment]
Hello everyone. I have the same situation here. Does anyone know how to fix it?

**Comment by sawicpx** [comment]
I am having the same issue here.

**Comment by DeXtmL** [comment]
Same issue. This could happen in Phase 3 or 4, seems to be a dead-lock issue to me. For ppl having such issue, switch to linux or go back to an earlier version.

**Comment by MariusMihai82** [comment]
Hey guys, 
This worked for me - same irritations on 94% stück on phase 4 , trying to write on external drive.

1) Go to ‘Device Manager’
2) Right click on the destination drive (external in my case) - Proprieties
3) in ‘Policies’ - select ‘Better Performance’ instead of ‘Quick removal’((Default)

+++

1)Same ‘Device Manager’
2) Universal Serial Bus controllers
3) right click each one - Proprieties
4) ‘Power Management’ - deselect all

Hope it works for you guys, good luck and all the best.

M

**Comment by chiabug** [comment]
I see the same issue on one particular machine.  The machine uses Windows 10 and sata drives.  It always sticks at the same place in phase 4.  The machine is unresponsive when this occurs and requires a power cycle.  I have tried low level formatting of the drives and then reinstalling Windows and Chia.  Same result.  Extensive analysis has uncovered no solution.  I have not tried loading Linux on this box to see if it gets past the problem point.  I suspect this is a bug in the code that occurs only in specific hardware circumstances.

**Reply by 2tekno** [reply]
I have the same issue on the Linux box, Ubuntu 21.04, AMD Ryzen 3950x, 64Gb RAM, Nvme as a temp. drive. Chia V1.16. I have tried to use only cli - same issue. And no parallel plotting, only 1 plot. Just tested the RAM - no issues. Tried different nvme drive - didn't help.

**Reply by MariusMihai82** [reply]
I am plotting on 2 Nvme - one works , one g’ets stuck at 94 (the Big one unfortunatelly!!!) 
1) kingston 500gb - works
2) wd blue sn550 2tb - does not

problem occured again, yesterday both worked.

i have changed Nvme slots so the 2tb is in the fast top pcie...
Drama continues...
I plot on external Wd My Book

**Comment by chenjie** [comment]
The problem went away when I reinstalled my RAM sticks.

**Reply by MariusMihai82** [reply]
What do you mean ‘reinstall’ ?

**Reply by chenjie** [reply]
remove and install

**Reply by chenjie** [reply]
Update, it's been stable for two days since I reinstalled my memory.

**Comment by MariusMihai82** [comment]
Fixed (again)

It was apparently related to formating the plotting Nvme.

works after Quick Format.

**Reply by Guidob1000** [reply]
HI! how do you format the nvme disk?

**Comment by lyantrux** [comment]
Guys I just encountered this issue and I first disabled usb sleeping mode and then unplug replug the external drive and just saw it moved to 95%!

**Comment by Guidob1000** [comment]
> Plotting often stuck at beginning of phase 4, sometime for hours that I have to restart GUI. When is stuck, all other parallel plotting process will be stuck as well, regardless which phase they are in.
> 
> Starting phase 4/4: Write Checkpoint tables into "/Volumes/Passport_A/Temp/plot-k32-2021-04-29-14-48-0cc8a805a6873981d9c3dc5e2703848b1dbba95f8d965373af25914777c98986.plot.2.tmp" ... Fri Apr 30 05:00:48 2021
> Starting to write C1 and C3 tables
> Bucket 0 uniform sort. Ram: 3.251GiB, u_sort min: 1.125GiB, qs min: 0.281GiB.

Hi! Someone is having this problem with an internal disk??? becuase i have been reading and all have this problem but with an external disk. If someone has this problem with an internal and managed to fix it, please help me
